http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -49,17 +58,67 @@
       onMethod_ = {@VisibleForTesting})
   private boolean connectionIsOpen = false;
 
+  @Setter(
+      value = AccessLevel.PACKAGE,
+      onMethod_ = {@VisibleForTesting})
+  private HttpClient httpClient = HttpClient.newHttpClient();
+
   private final URI serverUri;
 
   private boolean closed = false;
 
+  private WebSocket client;
+
   @Getter(
       value = AccessLevel.PACKAGE,
       onMethod_ = {@VisibleForTesting})
-  @Setter(
-      value = AccessLevel.PACKAGE,
-      onMethod_ = {@VisibleForTesting})
-  private WebSocketClient client;
+  private final WebSocket.Listener webSocketListener =
+      new Listener() {
+        private final StringBuilder textAccumulator = new StringBuilder();
+
+        @Override
+        public void onOpen(final WebSocket webSocket) {
+          synchronized (queuedMessages) {
+            client = webSocket;
+            connectionIsOpen = true;
+            queuedMessages.forEach(
+                message ->
+                    client
+                        .sendText(message, true)
+                        .exceptionally(logWebSocketError("Failed to send queued text.")));
+            queuedMessages.clear();
+          }
+          webSocket.request(1);
+        }
+
+        @Override
+        public CompletionStage<?> onText(
+            final WebSocket webSocket, final CharSequence data, final boolean last) {
+          textAccumulator.append(data);
+          if (last) {
+            listener.messageReceived(textAccumulator.toString());
+            textAccumulator.setLength(0);
+          }
+          webSocket.request(1);
+          return null;
+        }
+
+        @Override
+        public CompletionStage<?> onClose(
+            final WebSocket webSocket, final int statusCode, final String reason) {
+          if (!reason.equals(CLIENT_DISCONNECT_MESSAGE)) {
+            log.severe("Connection to server closed: " + reason);
```

Comment:
The server will disconnect banned clients, this is probably not a case where we want to do logging. Likely we want to show an info message, probably just let the connection closed listener handle it.